catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/activity/rtl/RtlArabicObjectSizeUnitTest.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.activity.rtl;
+
+import android.content.res.Resources;
+import android.os.SystemClock;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.MainMenuActivity;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Locale;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.uiespresso.util.UiTestUtils.getResourcesString;
+import static org.catrobat.catroid.uiespresso.util.UiTestUtils.isRTL;
+import static org.hamcrest.Matchers.containsString;
+
+@RunWith(AndroidJUnit4.class)
+public class RtlArabicObjectSizeUnitTest {
+
+	private String kiloByteUnit = getResourcesString(R.string.KiloByte_short);
+	private Locale arabicLocale = new Locale("ar");
+
+	@Rule
+	public BaseActivityInstrumentationRule<MainMenuActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(MainMenuActivity.class);
+
+	@Before
+	public void setUp() throws Exception {
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void testUnitSizeStringForKB() {
+		assertTrue(isRTL(Locale.getDefault()));
+		assertEquals(arabicLocale.getDisplayLanguage(), Locale.getDefault().getDisplayLanguage());
+		onView(withId(R.id.main_menu_button_programs))
+				.perform(click());
+		openActionBarOverflowOrOptionsMenu(baseActivityTestRule.getActivity());
+		onView(withText(R.string.delete))
+				.perform(click());
+		onView(withText(R.string.select_all))
+				.perform(click());
+		onView(withId(Resources.getSystem().getIdentifier("action_mode_close_button", "id", "android")))
+				.perform(click());
+		onView(withId(android.R.id.button1))
+				.perform(click());
+		SystemClock.sleep(3500);
```

Comment:
again, no sleeps.
if you have to deal with MainMenuActivity please have a look at the SmokeTest and how to use the idling resource. (ive already commented in more detail about idling resource and the problem with our MainMenuActivity on another PR i think)

Reply:
as i commented before, 

> sleep is needed to pass the Test, after deleting all the Programs, the App takes about 3500ms 

> to init "My first program".